core_impl/src/main/java/io/opencensus/implcore/tags/CurrentTaggingState.java

```diff
@@ -29,13 +31,25 @@
  */
 @ThreadSafe
 public final class CurrentTaggingState {
-  private volatile TaggingState currentState = TaggingState.ENABLED;
 
-  public TaggingState get() {
+  @GuardedBy("this")
+  private TaggingState currentState = TaggingState.ENABLED;
+
+  @GuardedBy("this")
+  private boolean isRead;
+
+  public synchronized TaggingState get() {
```

Comment:
I think it is fine for the moment, but I would like to create an issue to remove this global lock.